src/main/java/com/byclosure/webcat/stepdefinitions/helpers/SeleniumInteractions.java

```diff
@@ -0,0 +1,192 @@
+            throw new NoSuchElementException("\""+ locator +"\" not found");
+        }
+    }
+    
+    private WebElement findLinkByText(String text) {
+        final List<WebElement> elements = driver.findElements(By.linkText(text));
+        return elements.size() > 0 ? elements.get(0) : null;
+    }
+    
+    private WebElement findLinkByTitle(String title) {
+        final List<WebElement> elementsByTitle = driver.findElements(By.cssSelector("a[title='" + title + "']"));
+        return elementsByTitle.size() > 0 ? elementsByTitle.get(0) : null;
+    }
+    
+    private WebElement findLinkByImageAlt(String alt) {
+        final List<WebElement> elementsByImageAlt = driver.findElements(By.cssSelector("a > img[alt='" + alt + "']"));
+        return elementsByImageAlt.size() > 0 ? elementsByImageAlt.get(0) : null;
+    }
+
+    private WebElement findInputByText(String locator) {
+        final List<WebElement> labels = driver.findElements(By.cssSelector("label"));
+        for(WebElement l : labels) {
+            if(locator.equals(l.getText())) {
+                String inputId = l.getAttribute("for");
+                
+                if(null == inputId || "".equals(inputId)) {
+                    return null;
+                } else {
+                    final List<WebElement> inputs = driver.findElements(By.id(inputId));
+                    return inputs.size() > 0 ? inputs.get(0) : null;
+                }
+            }
+        }
+        return null;
+    }
+    
+    private WebElement findElementByName(String name){
+        final List<WebElement> elementsByName =  driver.findElements(By.name(name));
+        return elementsByName.size() > 0 ? elementsByName.get(0) : null;
+    }
+    
+    private WebElement findButtonByValue(String value) {
+        final List<WebElement> elements = driver.findElements(By.cssSelector("input[type='button'][value='" + value + "']"));
+        final List<WebElement> submitButtons = driver.findElements(By.cssSelector("input[type='submit'][value='" + value + "']"));
+        final List<WebElement> buttons = driver.findElements(By.cssSelector("button[value='" + value + "']"));
+        
+        elements.addAll(submitButtons);
+        elements.addAll(buttons);
+        
+        return elements.size() > 0 ? elements.get(0) : null;
+    }
+    
+    private WebElement findButtonByTitle(String title) {
+        final List<WebElement> elements = driver.findElements(By.cssSelector("input[type='button'][title='" + title + "']"));
+        final List<WebElement> submitButtons = driver.findElements(By.cssSelector("input[type='submit'][title='" + title + "']"));
+        final List<WebElement> buttons = driver.findElements(By.cssSelector("button[title='" + title + "']"));
+        
+        elements.addAll(submitButtons);
+        elements.addAll(buttons);
+        
+        return elements.size() > 0 ? elements.get(0) : null;
+    }
+    
+    private WebElement findElementById(String id) {
```

Comment:
This method is used because the webdriver's one throws an exception. We want to get the first element, but without throwing an exception if the element doesn't exist.